### DISCUSSION TITLE
运行的官方demo效果这么差么，这识别的都是啥呀，还是我的用法不对

### DISCUSSION THREAD

**Discussion by SWHL** [PaddlePaddle/PaddleOCR#13113] [Q&A]
### Discussed in https://github.com/PaddlePaddle/PaddleOCR/discussions/13049

<div type='discussions-op-text'>

<sup>Originally posted by **WusamX** June 12, 2024</sup>
<img width="793" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/91404cad-083c-4a49-af32-e057c67e8104">
<img width="692" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/88de79f3-05b8-4dce-8f6e-1b1d2c0d3764">
<img width="662" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/fd1aca42-160f-486c-8365-8349272792fd">
<img width="935" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/5ccd9942-e75b-49b8-adf3-337296dbabc6">
<img width="356" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/9c463d5f-80b7-4738-a64d-3529dc2ec688">
<img width="1129" alt="image" src="https://github.com/PaddlePaddle/PaddleOCR/assets/20549190/073a8529-c0f4-42bd-8382-60e3094e3c5c">
</div>

**Comment by GreatV** [comment]
估计是哪个地方方向分类器没有同步。

**Comment by jingsongliujing** [comment]
image_orientation=True 改为 image_orientation=False ，这个方向分类有问题paddlecls
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/45508593/55c556e2-7a52-42ad-969d-15943a516340)   ，
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/45508593/f570e350-9021-4f55-a541-f20638547bab)

**Comment by jingsongliujing** [comment]
其实是你用了方向分类，然后巧合的是刚好图像分类错了，就可视化的时候，你先旋转180度，再画这些可视化的框，就对上了
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/45508593/ca1b43af-54ab-4a6b-99a0-024b8cb322c3)
```python
import os
import cv2
from paddleocr import PPStructure, draw_structure_result, save_structure_res
from PIL import Image

def Structure_analysis(img_path):
    table_engine = PPStructure(show_log=True, image_orientation=True)
    save_folder = './output/table'
    img = cv2.imread(img_path)
    result = table_engine(img)
    save_structure_res(result, save_folder, os.path.basename(img_path).split('.')[0])

    for line in result:
        line.pop('img')
        print(line)

    font_path = './doc/fonts/simfang.ttf'  # PaddleOCR下提供字体包
    image = Image.open(img_path).convert('RGB')
    image=image.rotate(180)  # 旋转图片
    im_show = draw_structure_result(image, result, font_path=font_path)
    im_show = Image.fromarray(im_show)
    im_show.save('result.jpg')
    pass

if __name__ == '__main__':
    Structure_analysis('./ppstructure/docs/table/1.png')
    pass

**Comment by jingsongliujing** [comment]
但是奇怪的是文字识别好像有点问题

**Comment by GreatV** [comment]
@jingsongliujing  这个图左右翻转了，所以文字识别不对。

**Comment by D715-ky** [comment]
> 但是奇怪的是文字识别好像有点问题

您好，请问ppocr原本分类器的问题怎么改呢？我的任务中分类器还是需要的，但是使用默认的分类器就会出现镜像或者翻转识别了

**Comment by D715-ky** [comment]
> @jingsongliujing 这个图左右翻转了，所以文字识别不对。

您好，请问使用ppocr原本的分类器时，会将图像翻转，不过不是全部的子图都翻转，有一小部分会这样，然后识别结果就不对了类似于镜像识别，但是我的任务中分类器还是需要的，所以这种问题该怎么解决呢？

**Comment by jingsongliujing** [comment]
> > @jingsongliujing 这个图左右翻转了，所以文字识别不对。
> 
> 您好，请问使用ppocr原本的分类器时，会将图像翻转，不过不是全部的子图都翻转，有一小部分会这样，然后识别结果就不对了类似于镜像识别，但是我的任务中分类器还是需要的，所以这种问题该怎么解决呢？

这个就是模型精度的问题，官方模型可能针对特定场景的图片可能精度没那么高，建议你使用paddlecls里面的模型试试，实在不行，可以使用cls里的预训练模型再标注一版自己场景的数据进行训练：https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/docs/zh_CN/models/PULC/PULC_text_image_orientation.md

**Comment by D715-ky** [comment]
> > > @jingsongliujing 这个图左右翻转了，所以文字识别不对。
> > 
> > 
> > 您好，请问使用ppocr原本的分类器时，会将图像翻转，不过不是全部的子图都翻转，有一小部分会这样，然后识别结果就不对了类似于镜像识别，但是我的任务中分类器还是需要的，所以这种问题该怎么解决呢？
> 
> 这个就是模型精度的问题，官方模型可能针对特定场景的图片可能精度没那么高，建议你使用paddlecls里面的模型试试，实在不行，可以使用cls里的预训练模型再标注一版自己场景的数据进行训练：https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/docs/zh_CN/models/PULC/PULC_text_image_orientation.md

非常感谢，我用的ch_ppocr_moblie_v2.0_cls_infer，初始化在了PaddleOCR()里，可能是不太适应于我的数据场景，还有两个疑惑：（1）如果我自己构建文本方向分类数据集训练一个方向分类器，分了0,90和180度，还需要修改其他哪里才可以应用分类器和识别模型呢？（2）分类器判断出图像角度之后，是做了一些旋转操作转正图像的方向吗，是怎么操作的，在代码的哪个位置呢？

**Comment by jingsongliujing** [comment]
你就不用官方的文本方向分类，用自己的方向分类模型把图片矫正再进行ocr即可



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年6月18日(周二) 上午9:27
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [PaddlePaddle/PaddleOCR] 运行的官方demo效果这么差么，这识别的都是啥呀，还是我的用法不对 (Issue #13066)




    
@jingsongliujing 这个图左右翻转了，所以文字识别不对。
  
您好，请问使用ppocr原本的分类器时，会将图像翻转，不过不是全部的子图都翻转，有一小部分会这样，然后识别结果就不对了类似于镜像识别，但是我的任务中分类器还是需要的，所以这种问题该怎么解决呢？
  
这个就是模型精度的问题，官方模型可能针对特定场景的图片可能精度没那么高，建议你使用paddlecls里面的模型试试，实在不行，可以使用cls里的预训练模型再标注一版自己场景的数据进行训练：https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/docs/zh_CN/models/PULC/PULC_text_image_orientation.md
  
非常感谢，我用的ch_ppocr_moblie_v2.0_cls_infer，初始化在了PaddleOCR()里，可能是不太适应于我的数据场景，还有两个疑惑：（1）如果我自己构建文本方向分类数据集训练一个方向分类器，分了0,90和180度，还需要修改其他哪里才可以应用分类器和识别模型呢？（2）分类器判断出图像角度之后，是做了一些旋转操作转正图像的方向吗，是怎么操作的，在代码的哪个位置呢？

**Comment by D715-ky** [comment]
> 你就不用官方的文本方向分类，用自己的方向分类模型把图片矫正再进行ocr即可
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2024年6月18日(周二) 上午9:27 收件人: ***@***.***&gt;; 抄送: ***@***.******@***.***&gt;; 主题: Re: [PaddlePaddle/PaddleOCR] 运行的官方demo效果这么差么，这识别的都是啥呀，还是我的用法不对 (Issue #13066) @jingsongliujing 这个图左右翻转了，所以文字识别不对。 您好，请问使用ppocr原本的分类器时，会将图像翻转，不过不是全部的子图都翻转，有一小部分会这样，然后识别结果就不对了类似于镜像识别，但是我的任务中分类器还是需要的，所以这种问题该怎么解决呢？ 这个就是模型精度的问题，官方模型可能针对特定场景的图片可能精度没那么高，建议你使用paddlecls里面的模型试试，实在不行，可以使用cls里的预训练模型再标注一版自己场景的数据进行训练：https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/docs/zh_CN/models/PULC/PULC_text_image_orientation.md 非常感谢，我用的ch_ppocr_moblie_v2.0_cls_infer，初始化在了PaddleOCR()里，可能是不太适应于我的数据场景，还有两个疑惑：（1）如果我自己构建文本方向分类数据集训练一个方向分类器，分了0,90和180度，还需要修改其他哪里才可以应用分类器和识别模型呢？（2）分类器判断出图像角度之后，是做了一些旋转操作转正图像的方向吗，是怎么操作的，在代码的哪个位置呢？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

嗯呢  好的，我再训练个方向分类器，非常感谢Thanks♪(･ω･)ﾉ
